lib/utils/appList.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const url = require('url');
+const R = require('ramda');
+const scriptData = require('./scriptData');
+
+/*
+ * Map the MAPPINGS object, applying each field spec to the parsed data.
+ * If the mapping value is an array, use it as the path to the extract the
+ * field's value. If it's an object, extract the value in object.path and pass
+ * it to the function in object.fun
+*/
+
+const MAPPINGS = {
+  title: [2],
+  appId: [12, 0],
+  url: {
+    path: [9, 4, 2],
+    fun: (path) => new url.URL(path, 'https://play.google.com').toString()
+  },
+  icon: [1, 1, 0, 3, 2],
+  developer: [4, 0, 0, 0],
+  developerId: {
+    path: [4, 0, 0, 1, 4, 2],
+    fun: extaractDeveloperId
+  },
+  priceText: {
+    path: [7, 0, 3, 2, 1, 0, 2],
+    fun: (price) => price === undefined ? 'FREE' : price
+  },
+  free: {
+    path: [7, 0, 3, 2, 1, 0, 2],
+    fun: (price) => price === undefined
+  },
+  summary: [4, 1, 1, 1, 1],
+  scoreText: [6, 0, 2, 1, 0],
+  score: [6, 0, 2, 1, 1]
+};
+
+function extaractDeveloperId (link) {
+  const q = url.parse(link, {parseQueryString: true});
+  return q.query['id'];
+}
+
+function extractor (root) {
```

Comment:
I find a bit confusing that you have both extractor _and_ extract and both are exported. I would prefer to have just one of them exported, and with a short comment explaining what it does.